### ISSUE TITLE
Label's center alignment does not use stylebox margin

### ISSUE THREAD

**Issue by timothyqiu** [godotengine/godot#76202]
### Godot version

4.0.2.stable

### System information

Arch Linux

### Issue description

Lable's horizontal center alignment ignores stylebox margins.

Other alignment modes including vertical center alignment use stylebox margins.

In this screenshot, the background stylebox's content margin is 24, 24, 0, 0 (left, top, right, bottom). Texts in the center column should shift to the right.

![ksnip_20230418-171052](https://user-images.githubusercontent.com/372476/232730868-42676a22-7e66-4da5-a66c-55f444378ed8.png)

### Steps to reproduce

1. Create a Label, put some text, set it to center horitonally.
2. Add a stylebox to the label and set its content margin asymmetrical horizontally.

### Minimal reproduction project

[test-4.zip](https://github.com/godotengine/godot/files/11259982/test-4.zip)

**Comment by YuriSizov** [comment]
That's actually an interesting point. Should alignment be done based on the bounding box or on the content box? I think you can argue for either.

**Comment by timothyqiu** [comment]
In all the 8 alignment modes, only horizontal center aligns based on the bounding box, other 7 modes align based on the content box.

I think horizontal center assumes that left and right margins are always equal, so they are not taken into account.

**Comment by YuriSizov** [comment]
> In all the 8 alignment modes, only horizontal center aligns based on the bounding box, other 7 modes align based on the content box.

Not necessarily. For top, left, right, and bottom alignments and margins you will not see any difference whether we use the content box or the bounding box for the alignment. Whichever it is, we still add content margins to offset the content, so the end result is the same.

Only center alignments would have a noticeable effect. If we use the bounding box, then the content will be aligned to the visual boundaries of the control, so it always appears perfectly centered. If that content reaches one of the sides, the offsets would prevent it from getting too close to that side. If we use the content box, then the content will be aligned to the invisible content boundaries, resulting in it looking off-center if you use different margin values for the opposite sides.

There are two different use cases, and it's not clear to me which would be preferred. I agree that vertical and horizontal center alignments should be consistent though. The question is, which behavior makes the most sense. Or, alternatively, if we should allow both with some setting.
